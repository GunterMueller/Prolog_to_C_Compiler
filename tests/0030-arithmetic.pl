main :-
	Y = 3,
	X1 is Y + 2, display(X1), nl,
	X2 is Y - 2, display(X2), nl,
	X3 is Y * 2, display(X3), nl,
	X4 is Y / 2, display(X4), nl,
	X5 is Y // 2, display(X5), nl,
	(X1 =:= Y -> display('yes\n'); display('no\n')),
	(X1 =:= X1 -> display('yes\n'); display('no\n')),
	(X1 =\= Y -> display('yes\n'); display('no\n')),
	(X1 =\= X1 -> display('yes\n'); display('no\n')),
	(X1 > Y -> display('yes\n'); display('no\n')),
	(X1 < Y -> display('yes\n'); display('no\n')),
	(X1 >= Y -> display('yes\n'); display('no\n')),
	(X1 >= X1 -> display('yes\n'); display('no\n')),
	(X1 =< Y -> display('yes\n'); display('no\n')),
	(X1 =< X1 -> display('yes\n'); display('no\n')).
