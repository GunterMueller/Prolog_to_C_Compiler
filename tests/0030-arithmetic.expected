5
1
6
1.5
1
no
yes
yes
no
yes
no
yes
yes
no
yes
